File: credentials-verification-android/app/src/main/java/io/iohk/cvp/grpc/AsyncTaskResult.java
package io.iohk.cvp.grpc;

public class AsyncTaskResult<A> {

  private A result;
  private Exception error;

  public A getResult() {
    return result;
  }

  public Exception getError() {
    return error;
  }

  public AsyncTaskResult(A result) {
    this.result = result;
  }

  public AsyncTaskResult(Exception error) {
    this.error = error;
  }
}

File: credentials-verification-android/app/src/main/java/io/iohk/atala/prism/app/grpc/AsyncTaskResult.kt
package io.iohk.atala.prism.app.grpc

class AsyncTaskResult<A> {
    var result: A? = null
        private set
    var error: Exception? = null
        private set

    constructor() {}
    constructor(result: A) {
        this.result = result
    }

    constructor(error: Exception?) {
        this.error = error
    }
}
